### PR TITLE
Fixed copywrite errors in LG Ez and Bz.

### PR DESCRIPTION
**Pull request by toma-vlad**
I refactored the code that computes `Ez` and `Bz` for Gauss and Laguerre-Gauss. Added a check for `m = 0` and `p = 0` edge-cases and corrected some missing terms inside the formulae.

### CODE REVIEW

File: test/laguerre-gauss.jl [hunk 1]
```diff
@@ -4,6 +4,8 @@ using StaticArrays
 using UnitfulAtomic
 using LinearAlgebra
 
+pochhammer(a::Int,b::Int) = factorial(a+b-1)/factorial(a-1)
```
Comment (SebastianM-C) [inline test/laguerre-gauss.jl:7]: You can use the function defined in `LaserTypes` with `using LaserTypes: pochammer`

**Review by SebastianM-C — APPROVED** [review]
Thank you
